### ISSUE TITLE
Add option to generate validation set automatically as held-out data

### ISSUE THREAD

**Issue by emjotde** [amunmt/marian#414]
Add option to generate validation set automatically as held-out data, maybe as

```
--valid-heldout 5000
```
which could replace `--valid-sets ...`. This should sample from training data via reservoir sampling (see `sentencepiece_vocab.cpp` for an implementation) and save to a temporary file. Should be deterministic with given seed.

**Comment by emjotde** [comment]
Maybe also option to save the valid set to a given filename, with path as another option to the same option:

```
--valid-heldout 5000 path/to/saved/valid.prefix.
```

which depending on the number of input streams would save to `path/to/saved/valid.prefix.1 ... path/to/saved/valid.prefix.N`
